Makefile

```diff
@@ -1,104 +1,38 @@
-.PHONY: rel deps test show_test_results
+.PHONY: rel
 LOG=$(subst TARGET,$@,TARGET.log 2>&1 || (cat TARGET.log; exit 1))
 SILENCE_COVER =  | grep -v "logs.*\\.coverdata"
 SILENCE_COVER += | grep -v "Analysis includes data from imported files"
 SILENCE_COVER += | grep -v "WARNING: Deleting data for module"
 LOG_SILENCE_COVER=$(subst TARGET,$@,TARGET.log 2>&1 || (cat TARGET.log $(SILENCE_COVER); exit 1))
 EJABBERD_DIR = apps/ejabberd
-EJD_INCLUDE = $(EJABBERD_DIR)/include
-EJD_PRIV = $(EJABBERD_DIR)/priv
 XEP_TOOL = tools/xep_tool
 EJD_EBIN = $(EJABBERD_DIR)/ebin
-DEVNODES = node1 node2 node3 fed1
-
-all: deps compile
-
-compile: rebar
-	./rebar $(OPTS) compile > $(LOG)
-
-deps: rebar
-	./rebar get-deps > $(LOG)
-
-clean: rebar configure.out
-	rm -rf apps/*/logs
-	. ./configure.out && ./rebar clean
-
-quick_compile: rebar
-	./rebar $(OPTS) compile skip_deps=true > $(LOG)
-
-reload: quick_compile
-	@E=`ls ./rel/mongooseim/lib/ | grep ejabberd-2 | sort -r | head -n 1` ;\
-	rsync -uW ./apps/ejabberd/ebin/*beam ./rel/mongooseim/lib/$$E/ebin/ ;\
-
-reload_dev: quick_compile
```

Comment:
By
> and MongooseIM's beam file[s]

I mean this:
```
ls -l _build/mim1/rel/mongooseim/lib/ejabberd-2.1.8+mim*
lrwxr-xr-x  1 michal.piotrowski  staff  87 Dec 21 16:45 _build/mim1/rel/mongooseim/lib/ejabberd-2.1.8+mim-2.0.0-322-g5af94960 -> /Users/michal.piotrowski/projects/MongooseIM-rebar3-michalwski/_build/mim1/lib/ejabberd
```

The ejabberd app inside the release is symlinked to ejabberd inside mim1 profile. I've just checked it and doing just:

```
./rebar3 as mim1 compile
```

Recompiles modules and puts them in `_build/mim1/lib/ejabberd`

```
ls -lt _build/mim1/lib/ejabberd/ebin | head -10
total 9304
-rw-r--r--  1 michal.piotrowski  staff    6860 Dec 21 17:29 ejabberd.app
-rw-r--r--  1 michal.piotrowski  staff    4952 Dec 21 17:29 ejabberd.beam
-rw-r--r--  1 michal.piotrowski  staff   55420 Dec 21 17:29 ejabberd_s2s_in.beam
-rw-r--r--  1 michal.piotrowski  staff   91928 Dec 21 17:29 ELDAPv3.beam
-rw-r--r--  1 michal.piotrowski  staff    7484 Dec 21 17:29 XmppAddr.beam
-rw-r--r--  1 michal.piotrowski  staff   50268 Dec 21 17:29 eldap.beam
-rw-r--r--  1 michal.piotrowski  staff   12176 Dec 21 16:45 acl.beam
-rw-r--r--  1 michal.piotrowski  staff    9460 Dec 21 16:45 adhoc.beam
-rw-r--r--  1 michal.piotrowski  staff   16772 Dec 21 16:45 amp.beam
```

This is still slower than reload_dev, but with rebar3 where there is no `skip_deps` (at least I'm not able to do this) we probably can't be faster than this.

I'm not against re-adding `reload_dev`, I'm just trying to find if we can achieve the same with what we have built-in into rebar3.